### ISSUE TITLE
joins doesn't work with distinct after 1.5.1 upgrade

### ISSUE THREAD

**Issue by iercan** [sparklyr/sparklyr#2865]
Hi. 
We are using sparklyr 1.4.0 with spark 3.0.0. After upgrading sparklyr to 1.5.1 some of our codes started failing. After some investigation I realized if distinct is used with join, operation is failing. It can be reproduced with below code sample. 

```r
a <- sdf_len(sc,1000)
b <- sdf_len(sc,1000) %>% distinct()

a %>% left_join(b)  # left or right join doesn't matter. both are failing
```

It throw below error. 

```
Error: org.apache.spark.sql.AnalysisException: cannot resolve '`RHS.__row_num_33e4e28f_6d22_4a83_92b3_6e8b4635d5fb`' given input columns: [LHS.id, RHS.id]; line 1 pos 27;
'Project [id#3696 AS id#3706, 'RHS.__row_num_33e4e28f_6d22_4a83_92b3_6e8b4635d5fb AS __row_num_33e4e28f_6d22_4a83_92b3_6e8b4635d5fb#3707]
+- Join LeftOuter, (id#3696 = id#3705)
   :- SubqueryAlias LHS
   :  +- SubqueryAlias sparklyr_tmp_8ec16e1e_75e4_4827_b6c3_5e87069640f4
   :     +- LogicalRDD [id#3696], false
   +- SubqueryAlias RHS
      +- Project [id#3705]
         +- Sort [__row_num_33e4e28f_6d22_4a83_92b3_6e8b4635d5fb#3704 ASC NULLS FIRST], true
            +- Project [id#3705, __row_num_33e4e28f_6d22_4a83_92b3_6e8b4635d5fb#3704]
               +- SubqueryAlias dbplyr_105
                  +- Sort [__row_num_33e4e28f_6d22_4a83_92b3_6e8b4635d5fb#3704 ASC NULLS FIRST], true
                     +- Aggregate [id#3700], [first(__row_num_33e4e28f_6d22_4a83_92b3_6e8b4635d5fb#3703, false) AS __row_num_33e4e28f_6d22_4a83_92b3_6e8b4635d5fb#3704, first(id#3700, false) AS id#3705]
                        +- SubqueryAlias dbplyr_104
                           +- Project [id#3700, __row_num_33e4e28f_6d22_4a83_92b3_6e8b4635d5fb#3703]
                              +- Project [id#3700, __row_num_33e4e28f_6d22_4a83_92b3_6e8b4635d5fb#3703, __row_num_33e4e28f_6d22_4a83_92b3_6e8b4635d5fb#3703]
                                 +- Window [row_number() windowspecdefinition(null ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS __row_num_33e4e28f_6d22_4a83_92b3_6e8b4635d5fb#3703], [null ASC NULLS FIRST]
                                    +- Project [id#3700]
                                       +- SubqueryAlias sparklyr_tmp_e05e5eb3_5db0_4bd2_94de_fbb796b7bcf4
                                          +- LogicalRDD [id#3700], false

	at org.apache.spark.sql.catalyst.analysis.package$AnalysisErrorAt.failAnalysis(package.scala:42)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis$$anonfun$$nestedInanonfun$checkAnalysis$1$2.applyOrElse(CheckAnalysis.scala:143)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis$$anonfun$$nestedInanonfun$checkAnalysis$1$2.applyOrElse(CheckAnalysis.scala:140)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformUp$2(TreeNode.scala:333)
	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:72)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformUp(TreeNode.scala:333)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformUp$1(TreeNode.scala:330)
	at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$mapChildren$1(TreeNode.scala:399)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapProductIterator(TreeNode.scala:237)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:397)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:350)
	at org.apache.spark.sql.catalyst.trees.TreeNode.transformUp(TreeNode.scala:330)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.$anonfun$transformExpressionsUp$1(QueryPlan.scala:106)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.$anonfun$mapExpressions$1(QueryPlan.scala:118)
	at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:72)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.transformExpression$1(QueryPlan.scala:118)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.recursiveTransform$1(QueryPlan.scala:129)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.$anonfun$mapExpressions$3(QueryPlan.scala:134)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:238)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike.map(TraversableLike.scala:238)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:231)
	at scala.collection.immutable.List.map(List.scala:298)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.recursiveTransform$1(QueryPlan.scala:134)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.$anonfun$mapExpressions$4(QueryPlan.scala:139)
	at org.apache.spark.sql.catalyst.trees.TreeNode.mapProductIterator(TreeNode.scala:237)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.mapExpressions(QueryPlan.scala:139)
	at org.apache.spark.sql.catalyst.plans.QueryPlan.transformExpressionsUp(QueryPlan.scala:106)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis.$anonfun$checkAnalysis$1(CheckAnalysis.scala:140)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis.$anonfun$checkAnalysis$1$adapted(CheckAnalysis.scala:92)
	at org.apache.spark.sql.catalyst.trees.TreeNode.foreachUp(TreeNode.scala:177)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis.checkAnalysis(CheckAnalysis.scala:92)
	at org.apache.spark.sql.catalyst.analysis.CheckAnalysis.checkAnalysis$(CheckAnalysis.scala:89)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.checkAnalysis(Analyzer.scala:130)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.$anonfun$executeAndCheck$1(Analyzer.scala:156)
	at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.markInAnalyzer(AnalysisHelper.scala:201)
	at org.apache.spark.sql.catalyst.analysis.Analyzer.executeAndCheck(Analyzer.scala:153)
	at org.apache.spark.sql.execution.QueryExecution.$anonfun$analyzed$1(QueryExecution.scala:68)
	at org.apache.spark.sql.catalyst.QueryPlanningTracker.measurePhase(QueryPlanningTracker.scala:111)
	at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$1(QueryExecution.scala:133)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:763)
	at org.apache.spark.sql.execution.QueryExecution.executePhase(QueryExecution.scala:133)
	at org.apache.spark.sql.execution.QueryExecution.analyzed$lzycompute(QueryExecution.scala:68)
	at org.apache.spark.sql.execution.QueryExecution.analyzed(QueryExecution.scala:66)
	at org.apache.spark.sql.execution.QueryExecution.assertAnalyzed(QueryExecution.scala:58)
	at org.apache.spark.sql.Dataset$.$anonfun$ofRows$2(Dataset.scala:99)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:763)
	at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:97)
	at org.apache.spark.sql.SparkSession.$anonfun$sql$1(SparkSession.scala:606)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:763)
	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:601)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sparklyr.Invoke.invoke(invoke.scala:147)
	at sparklyr.StreamHandler.handleMethodCall(stream.scala:136)
	at sparklyr.StreamHandler.read(stream.scala:61)
	at sparklyr.BackendHandler.$anonfun$channelRead0$1(handler.scala:58)
	at scala.util.control.Breaks.breakable(Breaks.scala:42)
	at sparklyr.BackendHandler.channelRead0(handler.scala:39)
	at sparklyr.BackendHandler.channelRead0(handler.scala:14)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.ja
```

```
> utils::sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 9 (stretch)

Matrix products: default
BLAS:   /usr/lib/libblas/libblas.so.3.7.0
LAPACK: /usr/lib/lapack/liblapack.so.3.7.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] httr_1.4.1            forcats_0.5.0         readr_1.3.1           tibble_2.1.3          tidyverse_1.3.0       digest_0.6.25         R6_2.4.0              RMySQL_0.10.17       
 [9] DBI_1.1.0             assertthat_0.2.1      purrr_0.3.3           sparklyr.nested_0.0.3 jsonlite_1.6.1        rlang_0.4.8           reticulate_1.13       templates_0.3.0      
[17] snakecase_0.11.0      elasticsearchr_0.3.1  testthat_2.3.2        rlist_0.4.6.1         yaml_2.2.1            stringr_1.4.0         scales_1.0.0          mongolite_2.1.0      
[25] plotly_4.9.0          rprojroot_1.3-2       pool_0.1.4.3          sparklyr_1.5.1        ggplot2_3.2.1         sergeant_0.8.0        tidyr_1.0.0           data.table_1.12.8    
[33] dbplyr_1.4.2          dplyr_0.8.3           plyr_1.8.4            anomalize_0.2.0       lubridate_1.7.4       bigrquery_1.2.0       RMariaDB_1.0.6       

loaded via a namespace (and not attached):
 [1] nlme_3.1-144          fs_1.3.1              bit64_0.9-7           tools_3.6.3           backports_1.1.6       utf8_1.1.4            lazyeval_0.2.2        colorspace_1.4-1     
 [9] dat_0.4.0             withr_2.1.2           tidyselect_1.1.0      bit_1.1-15.2          compiler_3.6.3        rvest_0.3.5           cli_2.0.2             xml2_1.2.2           
[17] forge_0.2.0           askpass_1.1           rappdirs_0.3.1        rmarkdown_2.1         base64enc_0.1-3       pkgconfig_2.0.3       htmltools_0.4.0       fastmap_1.0.1        
[25] listviewer_3.0.0      htmlwidgets_1.5.1     readxl_1.3.1          rstudioapi_0.11       shiny_1.4.0           generics_0.0.2        aoos_0.5.0            config_0.3           
[33] magrittr_1.5          Formula_1.2-3         Matrix_1.2-18         Rcpp_1.0.2            munsell_0.5.0         fansi_0.4.0           lifecycle_0.1.0       stringi_1.4.3        
[41] grid_3.6.3            parallel_3.6.3        promises_1.1.0        crayon_1.3.4          lattice_0.20-40       haven_2.2.0           hms_0.5.2             flexdashboard_0.5.1.1
[49] knitr_1.25            pillar_1.4.2          uuid_0.1-4            reprex_0.3.0          glue_1.4.0            packrat_0.5.0         evaluate_0.14         modelr_0.1.6         
[57] vctrs_0.2.4           httpuv_1.5.2          cellranger_1.1.0      openssl_1.4.1         gtable_0.3.0          r2d3_0.2.3            xfun_0.10             mime_0.7             
[65] xtable_1.8-4          broom_0.5.5           later_1.0.0           viridisLite_0.3.0     ellipsis_0.3.0 
```

**Comment by yitao-li** [comment]
@iercan This was fixed in `sparklyr` 1.5.2 recently. Please run `install.packages("sparklyr")` again to install a version of `sparklyr` without this problem.

**Comment by iercan** [comment]
Oh. I didn't know new version is released. Thanks.
